dotnet/samples/Concepts/ChatCompletion/ChatHistoryReducers/ChatHistoryExtensions.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.ML.Tokenizers;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace ChatCompletion;
+
+/// <summary>
+/// Extension methods for <see cref="ChatHistory"/>."/>
+/// </summary>
+internal static class ChatHistoryExtensions
+{
+    private static readonly Tokenizer s_tokenizer = TiktokenTokenizer.CreateForModel("gpt-4");
+
+    /// <summary>
+    /// Returns the first system prompt from the chat history.
+    /// </summary>
+    internal static ChatMessageContent? GetSystemMessage(this ChatHistory chatHistory)
```

Comment:
It may be worth considering that multiple system prompts may exist.  In the agent framework, the system prompts / agent instructions are managed outside of the dynamic history....but for the cases you are addressing I agree it is in scope.  I think the cardinality is just off here.